src/permissions.rs

```diff
@@ -0,0 +1,111 @@
+//! Cross platform functions to restrict file permissions when using the file logger.
+#[cfg(unix)]
+use std::fs::File;
+use std::io;
+#[cfg(windows)]
+use std::path::Path;
+#[cfg(windows)]
+use winapi::um::winnt::{FILE_GENERIC_READ, FILE_GENERIC_WRITE, STANDARD_RIGHTS_ALL};
+
+/// This is the security identifier in Windows for the owner of a file. See:
+/// - https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/security-identifiers-in-windows#well-known-sids-all-versions-of-windows
+#[cfg(windows)]
+const OWNER_SID_STR: &str = "S-1-3-4";
+/// We don't need any of the `AceFlags` listed here:
+/// - https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-ace_header
+#[cfg(windows)]
+const OWNER_ACL_ENTRY_FLAGS: u8 = 0;
+/// Generic Rights:
+///  - https://docs.microsoft.com/en-us/windows/win32/fileio/file-security-and-access-rights
+/// Individual Read/Write/Execute Permissions (referenced in generic rights link):
+///  - https://docs.microsoft.com/en-us/windows/win32/wmisdk/file-and-directory-access-rights-constants
+/// STANDARD_RIGHTS_ALL
+///  - https://docs.microsoft.com/en-us/windows/win32/secauthz/access-mask
+#[cfg(windows)]
+const OWNER_ACL_ENTRY_MASK: u32 = FILE_GENERIC_READ | FILE_GENERIC_WRITE | STANDARD_RIGHTS_ALL;
+
+/// Function to set the umask of the log files to `600`.
+///
+/// This ensures the log files are not world-readable.
+#[cfg(unix)]
+pub fn restrict_file_permissions(file: File) -> io::Result<File> {
```

Comment:
How about aligning function signatures between Unix and Windows versions?

For example, I think we could do as follows:
```rust
#[cfg(unix)]
pub fn restrict_file_permissions<P: AsRef<Path>>(_path: P, file: File) -> io::Result<File> {
    ...abbrev...
    Ok(file)
}

#[cfg(windows)]
pub fn restrict_file_permissions<P: AsRef<Path>>(path: P, file: _File) -> io::Result<File> {
    ...abbrev...
    Ok(file)
}
```

By this change, the `cfg` branches in `src/file.rs` can be removed.

Reply:
In addition to the following comment, I thought it could be better to add another implementation for platforms other than Unix and Windows as follows:
```rust
#[cfg(not(any(unix, windows)))]
pub fn restrict_file_permissions<P: AsRef<Path>>(path: P, file: _File) -> io::Result<File> {
    Err(io::Error::new(io::ErrorKind::Other, "`restrict_permissions` feature is only available on Unix or Windows"))
}
```
By this addition, users, who even don't use Unix or Windows, could build `sloggers` without a "no such function" error.

Reply:
Yep good call. I've implemented both of these suggestions.